corelib-schemaorg/src/test/java/eu/europeana/corelib/edm/utils/SchemaOrgUtilsTest.java

```diff
@@ -64,7 +70,35 @@ public void toSchemaOrgEdmOrganizationTest() throws IOException {
         // If fails, then the field order definition misses some fields 
         assertEquals(expectedOutput, output);
     }
-    
+
+    @Test
+    public void testAddDateProperty() throws Exception {
+        FullBeanImpl bean = new FullBeanImpl();
+        MockFullBean.setTimespans(bean);
+        List<TimespanImpl> timespans= bean.getTimespans();
+        CreativeWork object = new CreativeWork();
+
+        //temporal coverage
+        SchemaOrgUtils.processDateValue(object, SchemaOrgConstants.PROPERTY_TEMPORAL_COVERAGE, timespans, false, "def", "1992-03-12"); //1
+        SchemaOrgUtils.processDateValue(object, SchemaOrgConstants.PROPERTY_TEMPORAL_COVERAGE, timespans, false, "def", "1824-1828"); //invalid
+        SchemaOrgUtils.processDateValue(object, SchemaOrgConstants.PROPERTY_TEMPORAL_COVERAGE, timespans, false, "def", "2002"); //2
+        SchemaOrgUtils.processDateValue(object, SchemaOrgConstants.PROPERTY_TEMPORAL_COVERAGE, timespans, false, "def", "18-02-1585"); //invalid
+        SchemaOrgUtils.processDateValue(object, SchemaOrgConstants.PROPERTY_TEMPORAL_COVERAGE, timespans, false, "def", "2017-07-26T01:00:00.000Z"); //3
+        SchemaOrgUtils.processDateValue(object, SchemaOrgConstants.PROPERTY_TEMPORAL_COVERAGE, timespans, false, "def", "http://semium.org/time/1977"); //4
+        SchemaOrgUtils.processDateValue(object, SchemaOrgConstants.PROPERTY_TEMPORAL_COVERAGE, timespans, false, "def", "http://semium.org/time/19xx"); //5
+        SchemaOrgUtils.processDateValue(object, SchemaOrgConstants.PROPERTY_TEMPORAL_COVERAGE, timespans, false, "def", "http://semium.org/time/1901"); //6
+        SchemaOrgUtils.processDateValue(object, SchemaOrgConstants.PROPERTY_TEMPORAL_COVERAGE, timespans, false, "def", "http://semium.org/time/1901"); //duplicate
+        SchemaOrgUtils.processDateValue(object, SchemaOrgConstants.PROPERTY_TEMPORAL_COVERAGE, timespans, false, "def", "23 march"); //invalid
+        SchemaOrgUtils.processDateValue(object, SchemaOrgConstants.PROPERTY_TEMPORAL_COVERAGE, timespans, true, "def", "23 testing invalid"); //invalid 7
+
+        assertNotNull(object.getTemporalCoverage());
+        assertEquals(object.getTemporalCoverage().size() , 7);
+        InputStream stream = getClass().getResourceAsStream(DATEFORMATCHECK_FILE);
```

Comment:
Please always close streams, even in tests

Reply:
done